code/modules/antagonists/changeling/powers/stalkingpresence.dm

```diff
@@ -0,0 +1,28 @@
+/datum/action/changeling/stalking_presence
+	name = "Stalking Presence"
+	desc = "We evolve a keen intution, able to detect the anxieties of nearby lifeforms."
+	helptext = "We will be able to detect the room, and direction of our prey as well as if they are injured."
+	button_icon_state = "stalking"
+	dna_cost = 1
+	power_type = CHANGELING_PURCHASABLE_POWER
+	menu_location = CLING_MENU_UTILITY
+
+
+/datum/action/changeling/stalking_presence/proc/valid_target(mob/target, mob/user)
+	return target.z == user.z && target.mind
+
+/datum/action/changeling/stalking_presence/sting_action(mob/user)
+	var/targets_by_name = list()
+	for(var/mob/living/carbon/human/H in GLOB.human_list)
+		if(!valid_target(H, user))
```

Comment:
Is there a reason you're making a new proc instead of just having the check in `valid_target` contained in this if statement?

Reply:
Turns out it REALLY does not like it if I don't.